lib/galaxy/jobs/runners/pulsar.py

```diff
@@ -265,6 +265,10 @@ def check_watched_item(self, job_state):
         if self.use_mq:
             # Might still need to check pod IPs.
             job_wrapper = job_state.job_wrapper
+            persisted_state = job_wrapper.get_state()
+            if persisted_state in model.Job.terminal_states + [model.Job.states.DELETED_NEW]:
```

Comment:
Should we bail if the state isn't running I wonder...

Reply:
Well it does need to stay in the monitor queue until it's running, but yeah it doesn't need to start checking ports until it's running.